editor/src/components/canvas/ui-jsx-canvas.tsx

```diff
@@ -1500,12 +1436,44 @@ function asErrorObject(e: unknown): Error {
   }
 }
 
-export class CanvasErrorBoundary extends React.PureComponent<CanvasErrorBoundaryProps> {
-  componentDidCatch(error: unknown, errorInfo: React.ErrorInfo) {
+interface CanvasErrorBoundaryState {
+  hasError: boolean
+}
+
+export class CanvasErrorBoundary extends React.Component<
+  CanvasErrorBoundaryProps,
+  CanvasErrorBoundaryState
+> {
+  constructor(props: CanvasErrorBoundaryProps) {
+    super(props)
+    this.state = { hasError: false }
+  }
+
+  static getDerivedStateFromError(_error: unknown): CanvasErrorBoundaryState {
+    return {
+      hasError: true,
+    }
+  }
+
+  componentDidUpdate(
+    prevProps: CanvasErrorBoundaryProps,
+    _prevState: CanvasErrorBoundaryState,
+  ): void {
+    if (prevProps.topLevelElementsIncludingScenes !== this.props.topLevelElementsIncludingScenes) {
+      // eslint-disable-next-line react/no-did-update-set-state
```

Comment:
now that componentDidUpdate is the safe recommended way of reacting to prop changes, does this eslint rule still help us? https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-did-update-set-state.md

Reply:
It's still a useful rule to have because the recommendation is to _only_ set the state _immediately_, and it must be wrapped in a condition. Having the rule will at the very least flag that the code has the potential to cause serious issues.